### PR TITLE
update datadog swagger examples to fix parameter name mismatch issue

### PR DESCRIPTION
**Pull request by kazrael2119**
customer met an issue https://github.com/Azure/azure-sdk-for-js/issues/27930 when using sdk samples which are generated by swagger examples.
It is because the parameter name mismatch. So I create a pr to fix this.

@vikotha could you help review this pr? thanks

### CODE REVIEW

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-08-01/extensions.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "swagger": "2.0",
   "info": {
-    "version": "2023-08-01",
+    "version": "2024-01-01",
```
[the remark this answers is not in the document]
Comment (kazrael2119) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-08-01/extensions.json]: revert it, It's not required

**Comment by vikotha** [comment]
LGTM

**Review by MaryGao — APPROVED** [review]
LGTM

**Comment by MaryGao** [comment]
Remove arm releted label considering only samples change.

**Pull request merged**